specs/AppLifecycle/Activation/AppLifecycle Activation.md

```diff
@@ -0,0 +1,248 @@
+    ActivationRegistrationManager registrationManager = 
+        AppLifecycle::GetActivationRegistrationManager();
+
+    // Register one or more supported image filetypes, an icon,
+    // and zero or more verbs for the File Explorer context menu.
+    registrationManager.RegisterForFileTypeActivation(
+        new string[]{ ".jpg", ".png", ".bmp" },
+        "MyResources.dll",
+        12,
+        new string[]{ "view", "edit" });
+
+    // Register another set of filetypes, with a different
+    // icon, and no specified verbs.
+    registrationManager.RegisterForFileTypeActivation(
+        new string[]{ ".mov", ".wmv", ".mp3" },
+        "MyResources.dll",
+        18);
+
+    // Register some URI schemes for protocol activation.
+    registrationManager.RegisterForProtocolActivation("foo");
+    registrationManager.RegisterForProtocolActivation("com.contoso.list");
+    registrationManager.RegisterForProtocolActivation("foo-bar");
+
+    // Register for startup activation.
+    registrationManager.RegisterForStartupActivation(
+        "MyTaskId", true, "Contoso Photo Viewer");
+
+    // Register for toast activation,
+    // where g_ToastGuid is defined as a static guid.
+    registrationManager.RegisterForToastActivation("Contoso Toast"); 
+}
+```
+### Get rich activation objects
+A Win32 app would typically get its command-line arguments very early in its WinMain, and we therefore expect the app to call the new AppLifecycle::GetActivatedEventArgs in the same place, instead of using the supplied lpCmdLine parameter or using GetCommandLineW. From there, the app can determine the specific ActivationKind, query the object for the specific XXXActivatedEventArgs type, and then access the properties and methods of that type.
+
+```c++
+void RespondToActivation()
+{
+    IActivatedEventArgs args = AppLifecycle::GetActivatedEventArgs();
+
+    if (args)
+    {
+        ActivationKind kind = args.Kind();
+        if (kind == ActivationKind::Launch)
+        {
+            auto launchArgs = args.as<LaunchActivatedEventArgs>();
+            DoSomethingWithLaunchArgs(launchArgs.Arguments());
+        }
+        else if (kind == ActivationKind::File)
+        {
+            auto fileArgs = args.as<FileActivatedEventArgs>();
+            DoSomethingWithFileArgs(fileArgs.Files());
+        }
+        else if (kind == ActivationKind::Protocol)
+        {
+            auto protocolArgs = args.as<ProtocolActivatedEventArgs>();
+            DoSomethingWithProtocolArgs(protocolArgs.Uri());
+        }
+        else if (kind == ActivationKind::StartupTask)
+        {
+            auto startupArgs = args.as<StartupTaskActivatedEventArgs>();
+            DoSomethingWithStartupArgs(startupArgs.TaskId());
+        }
+        else if (kind == ActivationKind::ToastNotification)
+        {
+            auto toastArgs = args.as<ToastNotificationActivatedEventArgs>();
+            DoSomethingWithToastArgs(toastArgs.Argument());
+        }
+    }
```

Comment:
Maybe put a comment down here like "The app was launched for other reasons, or was not launched for one of our known types, so let the regular app launch behavior handle it."

Reply:
The app can only be launched for the activation kinds it registers for (launch activation covers the case where it doesn't register for anything).